BUILD.gn

```diff
@@ -71,7 +71,7 @@ group("flutter") {
   }
 
   # If on the host, compile tools.
-  if (current_toolchain == host_toolchain) {
+  if (current_toolchain == host_toolchain || is_linux) {
```

Comment:
This condition no longer matches the comment. Aren't these host tools? It's not clear to me why you are cross-compiling them.

Reply:
Yes, I have to modify the comment. But I couldn't understand why `if (current_toolchain == host_toolchain)` was necessary. In order to run the Flutter SDK and Flutter Apps on ARM64 Linux platforms, we need to build `dart-sdk`, `frontend_server.dart.snapshot` and `const_finder.dart.snapshot` and so on. This condition is not necessary for cross build.

Reply:
I don't know the answer to that offhand.

Are they uploaded as artifacts and then downloaded and run by the `flutter` tool? If so, then you are correct that they aren't host tools, in which case we should instead be deciding when to build them based on support Flutter SDK environments.

Reply:
@stuartmorgan 
(cc: @christopherfujino please follow me.)

Thank you for your reply.

> Are they uploaded as artifacts and then downloaded and run by the flutter tool?

That's right.
I think that it is not necessary in the case where only cross build like Android and iOS, however it is necessary when running as Flutter SDK on Host. 

I think that all can be built for ARM64 without problems.

 #### //third_party/dart:create_sdk
 This is dart-sdk and download to `${INSTALL_PATH}/flutter/bin/cache/dart-sdk`. ARM64 is already supported ([link](https://github.com/dart-lang/sdk/wiki/Building-Dart-SDK-for-ARM-processors))

 #### //flutter/flutter_frontend_server:frontend_server
This is the Dart project, it can be built for ARM64.
It's downloaded to `${INSTALL_PATH}/flutter/bin/cache/artifacts/engine/linux-${arch}/frontend_server.dart.snapshot`

 #### //flutter/web_sdk
I'm not sure about this.

 #### //flutter/shell/testing
This is the C++ source code for `flutter_test`?
It's downloaded to `${INSTALL_PATH}/flutter/bin/cache/artifacts/engine/linux-${arch}/flutter_tester`

 #### //flutter/tools/const_finder
This is the Dart project.
It's downloaded to `${INSTALL_PATH}/flutter/bin/cache/artifacts/engine/linux-${arch}/const_finder.dart.snapshot`

 #### //flutter/tools/font-subset
This is the C++ source code.
It's downloaded to `${INSTALL_PATH}/flutter/bin/cache/artifacts/engine/linux-${arch}/font-subset`

Reply:
Makes sense; I thought "host" in the comment meant build host, rather that "Flutter host". I think you'll want `(is_linux && !is_chromeos)` though.

The comment should probably be something like:
```
# Flutter SDK artifacts should only be built when either doing host builds, or
# for cross-compiled desktop targets.
```

(I wouldn't mention Linux since macOS and/or Windows could be done this way in the future.)

Also, you should extract this to a variable like `build_engine_artifacts` since it's used in two places.

Reply:
Thank you so much. I fixed it. Could you review my change?